src/test/java/com/wikia/webdriver/elements/oasis/pages/InfoboxBuilderPage.java

```diff
@@ -0,0 +1,278 @@
+    wait.forElementVisible(rows.get(index));
+    rows.get(index).click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage addRowComponent() {
+    wait.forElementVisible(componentsButtons.get(0));
+    componentsButtons.get(0).click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage addTitleComponent() {
+    wait.forElementVisible(componentsButtons.get(1));
+    componentsButtons.get(1).click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage addImageComponent() {
+    wait.forElementVisible(componentsButtons.get(2));
+    componentsButtons.get(2).click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage deleteTitleComponentWithIndex(int index) {
+    wait.forElementVisible(titles.get(index));
+    titles.get(index).click();
+    wait.forElementVisible(deleteButton);
+    deleteButton.click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage deleteRowComponentWithIndex(int index) {
+    wait.forElementVisible(rows.get(index));
+    rows.get(index).click();
+    wait.forElementVisible(deleteButton);
+    deleteButton.click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage deleteImageComponentWithIndex(int index) {
+    wait.forElementVisible(images.get(index));
+    images.get(index).click();
+    wait.forElementVisible(deleteButton);
+    deleteButton.click();
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifyScrollbarIsVisible() {
+    Assertion.assertEquals(previewArea.getCssValue("overflow"), "auto");
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifyInfoboxPreviewBackgroundColor(String invocationBgColor) {
+    String previewBackgroundColor = getBackgroundColor();
+    Assertion.assertEquals(invocationBgColor, previewBackgroundColor);
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifyBackArrowFunctionality() {
+    wait.forElementVisible(backArrowButton);
+    backArrowButton.click();
+    Assertion.assertTrue(componentsButtons.get(0).isDisplayed());
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifyHelpDialog() {
+    wait.forElementVisible(questionMarkButton);
+    questionMarkButton.click();
+    Assertion.assertTrue(helpDialog.isDisplayed());
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifyTooltipOnHover() {
+    wait.forElementVisible(component.get(0));
+    builder.moveToElement(component.get(0)).perform();
+    Assertion.assertTrue(tooltip.isDisplayed());
+
+    return this;
+  }
+
+  /* Verifies default rendered template structure, which should contain:
+     1 title component, 1 image component and 2 row components
+  */
+  public InfoboxBuilderPage verifyDefaultTemplateStructure() {
+    Assertion.assertEquals(this.titles.size(), 1);
+    Assertion.assertEquals(this.images.size(), 1);
+    Assertion.assertEquals(this.rows.size(), 2);
+
+    return this;
+  }
+
+  public InfoboxBuilderPage verifySelectedComponentBorderStyle(int index) {
+    wait.forElementVisible(component.get(index));
+    component.get(index).click();
```

Comment:
Do we really want to do that? This will generate a line for almost every method. I'm not sure but I guess even Sonar highlights it to refactor.

Is my way of writing that is bad?